libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -255,7 +255,9 @@ void GCS_MAVLINK::ftp_worker(void) {
 
                         // sanity check that our the request looks well formed
                         const size_t file_name_len = strnlen((char *)request.data, sizeof(request.data));
-                        if ((file_name_len != request.size) || (request.size == 0)) {
+                        const bool is_req_size_consider_tnull = ((request.size - file_name_len == 1) &&
```

Comment:
please break out this check into a function check_string_name_length() or similar and use that in the 7 locations instead of copying the logic 7 times
thanks!